docs/security.md

```diff
@@ -0,0 +1,143 @@
+The full list of all configurations which apply for ssl are listed in [network configurations](configuration/network.md) - and namespaced under `celeborn.ssl`.
+ 
+{!
+include-markdown "./configuration/network-module.md"
+start="<!--begin-include-->"
+end="<!--end-include-->"
+!}
+
+Note that `celebord.ssl`, **without any module**, can be used to set SSL default values which applies to all modules.
+
+Also note that `data` module at application side, maps to `push` and `fetch` at worker - hence, for SSL configuration, worker configuration for `push` and `fetch` should be compatible with each other and with `data` at application side.
+
+### Example configuration
+
+#### Master/Worker configuration for TLS
+
+```properties
+
+# TLS configuration
+celeborn.ssl.rpc_service.enabled                true
+# Location of the java keystore which contains the private key and certificate for the master/worker.
+celeborn.ssl.rpc_service.keyStore               /mnt/disk1/celeborn/conf/ssl/server.jks
+celeborn.ssl.rpc_service.keyStorePassword       password
+# Location of the java truststore which contains the CA certs which can validate the master/worker certificate 
+celeborn.ssl.rpc_service.trustStore             /mnt/disk1/celeborn/conf/ssl/truststore.jks
+celeborn.ssl.rpc_service.trustStorePassword     changeit
+```
+
+#### Application configuration for TLS
+
+```properties
+
+
+# TLS
+
+# Configure rpc_app to enable ssl between lifecyclemanager and executors
+spark.celeborn.ssl.rpc_app.enabled                              true
+# Use auto ssl to generate a self-signed certificate at lifecyclemanager, to secure network communication.
+spark.celeborn.ssl.rpc_app.autoSslEnabled                       true
+
+# Secure communication with celeborn servers
+spark.celeborn.ssl.rpc_service.enabled                          true
+# trust store with CA certs, to verify the celeborn service certificate   
+spark.celeborn.ssl.rpc_service.trustStore                       /etc/ssl/certs/truststore.jks
+spark.celeborn.ssl.rpc_service.trustStorePassword               changeit
+```
+
+
+## Authentication
+
+Celeborn supports authentication to prevent unauthorized access or modifications to an application's data. 
+When enabled, lifecyclemanager registers the application with Celeborn service, and negotiates a shared secret.
+All further connections, from the application to Celeborn servers, will first be authenticated - and only an authorized connection can read/modify data for a registered application.
+
+The `shared secret`, which is generated as part of application registration, is used to authenticate all subsequent connections.
+Even though Celeborn does not transmit the secret, in the clear, as part of authentication - it still sends it in the clear during registration - and so enabling TLS for `rpc_service` and `rpc_app` transport modules is recommended (see above on how).  
+
+Note: SASL **requires use of internal port**.
+
+| Property Name | Default | Description |
```

Comment:
Is this table necessary? IMO, the document of config options is better to be unfied, no need to document again in other document? WDYT?

Reply:
This is currently not documented elsewhere.
+CC @otterc